src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs

```diff
@@ -940,7 +940,9 @@ private ParameterExpression HandleLambdaParameters(IEnumerable<RangeVariable> ra
             foreach (RangeVariable rangeVariable in rangeVariables)
             {
                 ParameterExpression parameter;
-                if (!_lambdaParameters.TryGetValue(rangeVariable.Name, out parameter))
+
+                // Create a Parameter Expression for rangeVariables which are not $it Lambda parameters or $this.
+                if (!_lambdaParameters.TryGetValue(rangeVariable.Name, out parameter) && rangeVariable.Name != ODataThisParameterName)
```

Comment:
In the comment you talk about creating parameter expression for range variables that are neither $it lambda parameters or $this but your change only added `&& rangeVariable.Name != ODataThisParameterName`. Will that suffice?

Reply:
This code `!_lambdaParameters.TryGetValue(rangeVariable.Name, out parameter)` ensures we don't create Parameter expressions for lambda parameters